### ISSUE TITLE
buildspec-- handle manifest validation failures

### ISSUE THREAD

**Issue by huanjani** [aws/copilot-cli#3072]
Currently, if a workload manifest fails the validation step, the cause for the build failure isn't immediately clear. We should try to fail faster or surface the error better.

**Comment by wolverian** [comment]
Just as an example, my experience today might illustrate why this is a problem. (If this isn't the issue you meant, @huanjani, my apologies and I can create a new issue.)

I changed my deployment to `platform: linux/arm64`. The build step of my pipeline passed, but the deployment failed with this unclear message:
<img width="623" alt="Screenshot 2021-11-25 at 14 24 11" src="https://user-images.githubusercontent.com/42124/143441302-9cbac322-ab27-458b-9c8b-da02fe3934df.png">

Looking into the execution details confused me even more!

<img width="971" alt="Screenshot 2021-11-25 at 14 24 21" src="https://user-images.githubusercontent.com/42124/143441424-c87efc65-1ccd-4044-aba9-b254a04f0146.png">

Here we see a message with a red background (it's an error!) saying the stack updated (it's not an error). So which one is it?

I downloaded the input to the deployment step and saw the deployment manifest (i.e. the CloudFormation template) was _empty_. I then looked into the Build step logs and found this error hidden among the thousands of lines of output:

```
✘ validate manifest against environment test: validate ARM: 'Fargate Spot' is not supported when deploying on ARM architecture
```

The build didn't stop on this error, though, so I had to go through this backwards debugging session to find out the root cause.
